File: simple-log/src/main/java/com/don11995/log/Group.java
/*
 * Modified by Vladyslav Lozytskyi on 11.04.18 1:37
 * Copyright (c) 2018. All rights reserved.
 */

package com.don11995.log;

import android.text.TextUtils;

import java.util.Locale;

public class Group {

    private String mGroupName;
    private String mText;
    private String mTag;

    /**
     * Create group for text and print it in one block
     *
     * @param group name of the group
     */
    public Group(String group) {
        mGroupName = group;
    }

    private void handleAppend(String text, Object... args) {
        if (!TextUtils.isEmpty(mText)) {
            mText += "\n\t";
        } else {
            mText = "";
        }
        if (args == null || args.length == 0) {
            mText += text;
        } else {
            mText += String.format(Locale.getDefault(), text, args);
        }
    }

    /**
     * Append new line to group
     *
     * @param text format text
     * @param args args to use with {@link String#format(String, Object...)}
     * @return current group
     */
    public Group append(String text, Object... args) {
        handleAppend(text, args);
        return this;
    }

    /**
     * Set custom tag to output in logcat
     *
     * @param tag custom tag for output log
     * @return current group
     */
    public Group tag(String tag) {
        mTag = tag;
        return this;
    }

    String getGroupName() {
        return mGroupName;
    }

    String getText() {
        return mText;
    }

    public String getTag() {
        return mTag;
    }
}

File: simple-log/src/main/java/com/don11995/log/Group.kt
/*
 * Modified by Vladyslav Lozytskyi on 05.05.18 22:36
 * Copyright (c) 2018. All rights reserved.
 */

package com.don11995.log

import java.util.*

/**
 * Create group for text and print it in one block
 *
 * @param mGroupName name of the group
 */
class Group(internal val mGroupName : String) {

    internal var mText: String = ""
    internal var mTag: String? = null

    private fun handleAppend(text: String, vararg args: Any) {
        if (!mText.isEmpty()) {
            mText += "\n\t"
        }
        mText += if (args.isEmpty()) {
            text
        } else {
            String.format(Locale.getDefault(), text, *args)
        }
    }

    /**
     * Append new line to group
     *
     * @param text format text
     * @param args args to use with [String.format]
     * @return current group
     */
    fun append(text: String, vararg args: Any): Group {
        handleAppend(text, *args)
        return this
    }

    /**
     * Set custom tag to output in logcat
     *
     * @param tag custom tag for output log
     * @return current group
     */
    fun tag(tag: String): Group {
        mTag = tag
        return this
    }
}